packages/sdk-cli/src/models/localRelay/relayInfo.test.ts

```diff
@@ -83,87 +100,79 @@ describe('pollRelayInfo', () => {
+    // readJsonFile() is called once as soon as pollRelayJson() is called
+    // hence we start counter from 1
+    for (let i = 1; i <= 2; i += 1) {
+      jest.advanceTimersByTime(100);
+      await flushPromises();
+      expect(readJsonFileSpy).toHaveBeenCalledTimes(i + 1);
+    }
 
-    expect(readJsonFileSpy).toHaveBeenCalledTimes(3);
+    await expect(poll).resolves.toEqual(relayInfo);
   });
 
-  /**
-   * With timeout 500ms, and interval 200ms, pollRelayInfo can only make max. 2 poll calls + 1 initial, at t = 0.
-   * We check whether polls are really called in the specified interval and not slower/faster (so exactly 3 times).
-   * setInterval/Timeout/etc. are not guaranteed to call functions at __exactly__ the time specified,
-   * so there is a margin of 100ms (2 x 200ms + 100ms).
-   */
+  // With timeout 500ms, and interval 200ms, pollRelayInfo can only make max. 2 poll calls + 1 initial, at t = 0.
+  // We check whether polls are really called in the specified interval and not slower/faster (so exactly 3 times).
   it('polls in regular intervals', async () => {
-    const timeout = 500;
-    const interval = 200;
-    const readJsonFileSpy = jest
-      .spyOn(util, 'readJsonFile')
-      .mockResolvedValue({});
-
-    await expect(pollRelayInfo(timeout, interval)).rejects.toThrow();
-
-    // Calls: 0, 1000, 2000, 3000
-    expect(readJsonFileSpy).toHaveBeenCalledTimes(
-      Math.floor(timeout / interval) + 1,
-    );
-
-    readJsonFileSpy.mockRestore();
-  });
-
-  /**
-   * With timeout 10 times larger than the interval, pollRelayInfo could be called 10 times.
-   * However, we set the first poll request's execution time at almost the timeline length – 800ms.
-   * The request is successful, so the interval should be cleared and value returned.
-   * Therefore, we check whether the interval is blocked whenever there is an existing request in-flight.
-   */
-  it('has only 1 in-flight poll request at a time', async () => {
-    const timeout = 1000;
+    const timeout = 10000;
     const interval = 100;
-    const pollExecTime = 800;
-
-    const value: RelayInfo = { port: 1, pid: 1 };
-    const readJsonFileSpy = jest.spyOn(util, 'readJsonFile').mockImplementation(
-      () =>
-        new Promise<RelayInfo>((resolve) =>
-          setTimeout(() => resolve(value), pollExecTime),
-        ),
-    );
+    readJsonFileSpy = jest.spyOn(util, 'readJsonFile').mockResolvedValue({});
 
-    await expect(pollRelayInfo(timeout, interval)).resolves.toEqual(value);
+    pollRelayInfo(timeout, interval);
+    await flushPromises();
 
-    expect(readJsonFileSpy).toHaveBeenCalledTimes(
-      Math.floor(timeout / pollExecTime),
-    );
-
-    readJsonFileSpy.mockRestore();
+    // Checking if the behaviour is correct a couple of times is sufficient
+    const reps = 2;
+    // We need to ensure the timeout & interval actually allow for "a couple" of runs without timing out
+    if (timeout / interval < reps) {
```

Comment:
This seems like the code of check that should be in the implementation, not the test, with the test checking for the throw?

Reply:
Perhaps flip it and calculate interval from repls so it's guaranteed to be correct rather than checking the test itself?

Reply:
https://github.com/Fitbit/developer-bridge/pull/176/commits/5361e00a6a230680cf9b361f1b418b5ebe087d37